### PR TITLE
Bitfinex Brokerage Updates

### PR DESCRIPTION
**Pull request by StefanoRaggi**
#### Description
- Replaced `WebSocketSharp` wrapper with `System.Net.WebSockets` wrapper
- Updated rate limits for multiple WebSocket connections
- Updated Bitfinex symbol list (new listings + delistings)

#### Motivation and Context
- Live trading stability

#### Requires Documentation Change
No.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.

### CODE REVIEW

File: Brokerages/Bitfinex/BitfinexSymbolMapper.cs [hunk 1]
```diff
@@ -91,7 +97,82 @@ public static List<Symbol> KnownSymbols
...
+            "DADUSD","DADBTC","DADETH",
+            "ORSBTC","ORSETH",
+            "AUCBTC","AUCETH",
+            "POYBTC","POYETH",
+            "FSNBTC","FSNETH",
+            "CBTBTC","CBTETH",
+            "ZCNBTC","ZCNETH",
+            "SENUSD","SENBTC","SENETH",
+            "NCABTC","NCAETH",
+            "CNDBTC",
```
Comment (Martin-Molinero) [inline Brokerages/Bitfinex/BitfinexSymbolMapper.cs:139]: Wonder whats the format logic behind having 1,2,3... symbols per line? Worth a comment I think

File: Brokerages/Bitfinex/BitfinexSubscriptionManager.cs [hunk 2]
```diff
@@ -204,6 +208,31 @@ private BitfinexWebSocketWrapper GetFreeWebSocket(BitfinexChannel channel)
...
+                webSocket.Connect();
+
+                if (!connectedEvent.WaitOne(ConnectionTimeout))
+                {
+                    throw new Exception("BitfinexSubscriptionManager.Connect(): WebSocket connection timeout.");
+                }
+            }
+            finally
+            {
+                webSocket.Open -= onOpenAction;
```
Comment (Martin-Molinero) [inline Brokerages/Bitfinex/BitfinexSubscriptionManager.cs:232]: nit - think could also dispose of `connectedEvent`

File: Brokerages/Bitfinex/BitfinexSymbolMapper.cs (hunk 1 above)
Comment (StefanoRaggi) [reply Brokerages/Bitfinex/BitfinexSymbolMapper.cs:139]: One coin can have one or more tradable pairs;
when updating this list I just found it less error-prone to use one line per coin.

Comment (Martin-Molinero) [reply Brokerages/Bitfinex/BitfinexSymbolMapper.cs:139]: Sounds good 👍. Couple of lines don't seem to respect it though: `"RRTBTC","QTMETH", "MNAETH","FUNBTC", "LRCETH","WAXETH",`

**Pull request merged**